generic-update-script.rb

```diff
@@ -142,6 +142,18 @@
   )
 end
 
+# Add additional credentials for repositories that needs authentication
+if ENV["PRIVATE_REPOSITORIES"]
+  feeds = JSON.parse("#{ENV["PRIVATE_REPOSITORIES"]}")
+  feeds.each do |child|
+    credentials << {
+      "type" => "#{child['type']}",
+      "url" => "#{child['url']}",
+      "token" => "#{child['token']}",
+    }
+  end
```

Comment:
Why not just `credentials += JSON.parse("#{ENV["PRIVATE_REPOSITORIES"]}")` (or an equivalent, with better error handling)? That way you don't have to make assumptions about the structure the credentials need to take.